Creating an exception handler with the `Context` argument

Seems it's not possible to do so with the proc macro.

```rust
error[E0425]: cannot find value `save_frame` in this scope
  --> /Users/mabez/Development/rust/embedded/projects/xtensa-atomic-emulation-trap/src/lib.rs:67:46
   |
67 | unsafe fn __exception(cause: ExceptionCause, save_frame: &mut Context) {
   |                                              ^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`
```

My work around for now is the following:

```rust
#[no_mangle] // TODO #[xtensa_lx_rt::exception] doesn't work
#[link_section = ".rwtext"]
unsafe fn __exception(cause: ExceptionCause, save_frame: &mut Context) {
```

Also longer term it might be good to be able to set a custom symbol? For instance, I want to create a custom handler in the atomic emulation crate, but still allow users to setup there own handler for handling non atomic emulation issues - any thoughts on this are welcome.

--- a/procmacros/src/lib.rs
+++ b/procmacros/src/lib.rs
@@ -213,7 +213,7 @@ pub fn exception(args: TokenStream, input: TokenStream) -> TokenStream {
         #[export_name = "__exception"]
         pub unsafe extern "C" fn #tramp_ident(
             cause: xtensa_lx_rt::exception::ExceptionCause,
-            frame: xtensa_lx_rt::exception::Context
+            frame: &mut xtensa_lx_rt::exception::Context
         ) {
             #ident(
                 #(#args),*

--- a/src/exception/esp32.rs
+++ b/src/exception/esp32.rs
@@ -68,9 +68,9 @@ pub struct Context {
 
 extern "Rust" {
     /// This symbol will be provided by the user via `#[exception]`
-    fn __exception(cause: ExceptionCause);
+    fn __exception(cause: ExceptionCause, save_frame: &mut Context);
     /// No attribute is supplied for this symbol as the double exception can hardly occur
-    fn __double_exception(cause: ExceptionCause);
+    fn __double_exception(cause: ExceptionCause, save_frame: &mut Context);
 
     /// This symbol will be provided by the user via `#[interrupt(1)]`
     fn __level_1_interrupt(level: u32, save_frame: &mut Context);

